### PR TITLE
Team cog

### PR DESCRIPTION
**Pull request by eluric**
Created:
- Team cog
- SQL queries for teams
- SQL queries for players

Users can now create teams, leave teams and invite players to their team.

### CODE REVIEW

File: cogs/team.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        # check invited user is not already in a team
+        if await player_query.get_player(str(invited_user.id)):
+            await interaction.response.send_message(
+                "The user you're trying to invite is already in a team.", ephemeral=True
+            )
+            return
+
+        # check that team is not full
+        if len(await team_query.get_team_members()) == MAX_TEAM_SIZE:
+            await interaction.response.send_message(
```
Comment (SkellyBG) [inline cogs/team.py:148]: missing return here?

File: cogs/team.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        """
+        Completely optional but we could make it so players can add
+        custom accept/reject messages.
+        """
+
+        # add invited user to the team and database
+        async def accept_callback(interaction: discord.Interaction):
+            new_player = interaction.user
+            team = await team_query.get_team(team_name)
```
Comment (SkellyBG) [inline cogs/team.py:173]: what happens if the team is removed after sending the invite, but before the user accept it?

Comment (SkellyBG) [reply cogs/team.py:173]: additionally, what if there are two or more invites send when say, there are 4 current people in a team? wouldn't it allow for more than 5 people in a team?

File: cogs/team.py (hunk 1 above)
Comment (eluric) [reply cogs/team.py:148]: Yes. Fixed

**Review by SkellyBG — APPROVED** [review]
LGTM!

**Pull request merged**